docs/exceptions.md

```diff
@@ -80,6 +97,9 @@ The code is considered impure when it's an `IO` action, a function that returns
 an `IO` action, or similar. The code can be _potentially_ impure when it is
 written in an abstract monad that can be instantiated to `IO`.
 
+*Comment:* bad definition. `foo = return 42` can be instantiated to `IO Int`
```

Comment:
I'd say some applicative term `x` is pure iff

```
fmap (const ()) x == pure ()
```

and impure otherwise.

For instance, every `Reader r t` term is pure, but not every `State s t`.

Reply:
This looks plausible. Mind submitting a separate PR with this improvement?